packages/gestalt/src/Masonry.js

```diff
@@ -589,6 +589,17 @@ export default class Masonry<T: { ... }> extends ReactComponent<Props<T>, State<
         .filter((item) => item && !measurementStore.has(item))
         .slice(0, itemsToMeasureCount);
 
+      // If there is a two column module be sure that we only calculate new positions
+      // for batch size number of items
+      const itemsToRender = hasTwoColumnItems
```

Comment:
this works since it effectively limits the number of items we use to calculate new positions, but the downside here is that it will take more render cycles to complete, depending on where the two column item is positioned.

example:
assuming 16 items, where the two col module is in position 9 (your example in the test):
- first pass render 6 items
- second pass render 6 additional items
- third pass render 6 additional items (this one has the two column module)
- fourth pass render the remaining 4

alternatively, what if we continue to render/layout all  items but leave it up to the layout on how many items it wants to factor into the graph by limiting the size of this array here: https://github.com/pinterest/gestalt/blob/0a517310803467aaf2bc1f017d88ad38405dee26/packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js#L266. This should achieve a similar effect since it limits the number of items that are factored into the graph but allow us to do a full layout sooner

Going off the previous example, we would render all 16 items in one pass but the two column layout would only take X amount of items into account